src/Configuration/src/SpringBoot/SpringBootEnvironmentVariableProvider.cs

```diff
@@ -61,11 +61,8 @@ public override void Load()
             {
                 string? value = Data[key];
 
-                if (value != null)
-                {
-                    string newKey = key.Contains('.') ? key.Replace('.', ':') : key;
-                    data[newKey] = value;
-                }
+                string newKey = key.Contains('.') ? key.Replace('.', ':') : key;
+                data[newKey] = value;
```

Comment:
This seems like it could be behavioral change. I'm not sure how I feel about it, is it fixing a bug?
Assuming we have something like `My.key.with.null.value=null`...
With the original behavior it looks like the key/value would disappear, but now `My:key:with:null:value=null` will make it into the final configuration?

Reply:
Yes, it's a breaking change in .NET 10, documented [here](https://learn.microsoft.com/en-us/dotnet/core/compatibility/extensions/10.0/configuration-null-values-preserved). I initially made it conditional on the TFM in [this commit](https://github.com/SteeltoeOSS/Steeltoe/pull/1615/commits/49ec3f790e77b1da41bd3a2ce2d37ad58e9a8282). In a later commit, I bumped `FoundationalVersion`, so now it applies unconditionally.